sympy/polys/matrices/sdm.py

```diff
@@ -94,6 +94,13 @@ def __rmul__(a, b):
         else:
             return NotImplemented
 
+    def __smul__(a, e):
+        rows, cols = a.shape
+        for r in range(rows):
+            for c in range(cols):
+                a[r][c] = a[r][c]*e
+        return a
```

Comment:
You seem to keep inventing special methods that don't exist. There is no special method called `__smul__`.

In any case this scalar multiplication should use `binop_dict` which is more efficient as it reduces hash-table lookups. It should also have a special check for multiplication by zero or one which can be implemented most efficiently.

Reply:
> You seem to keep inventing special methods that don't exist. There is no special method called `__smul__`.
> 
Sorry about that, will change it and keep it in mind for the future.

> In any case this scalar multiplication should use `binop_dict` which is more efficient as it reduces hash-table lookups.
>
I am not sure what is `binop_dict`. 

Does the method in `__getitem__` suggested by you can be implemented?
I mean to ask can we iterate this way and multiply with scalar?
```
for i, row in self.items():
    if i in ri:
        row = {ci.index(j): e for j, e in row.items() if j in ci}
        sdm[ri.index(i)] = row
```

Reply:
The `binop_dict` function is in the same file you are editing and is used by other methods. Actually this should be `unop_dict` since it's effecitvely an unary operation.

Reply:
Sure, thank you for the info.